lib/elixir/lib/calendar/datetime.ex

```diff
@@ -206,40 +319,167 @@ defmodule DateTime do
+  section in the module documentation.
 
   ## Examples
 
       iex> DateTime.from_naive!(~N[2016-05-24 13:26:08.003], "Etc/UTC")
       #DateTime<2016-05-24 13:26:08.003Z>
 
+      iex> DateTime.from_naive!(~N[2018-05-24 13:26:08.003], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      #DateTime<2018-05-24 13:26:08.003+02:00 CEST Europe/Copenhagen>
+
   """
   @doc since: "1.4.0"
-  @spec from_naive!(NaiveDateTime.t(), Calendar.time_zone()) :: t
-  def from_naive!(naive_datetime, time_zone) do
-    case from_naive(naive_datetime, time_zone) do
+  @spec from_naive!(
+          NaiveDateTime.t(),
+          Calendar.time_zone(),
+          Calendar.time_zone_database()
+        ) :: t
+  def from_naive!(
+        naive_datetime,
+        time_zone,
+        time_zone_database \\ Calendar.get_time_zone_database()
+      ) do
+    case from_naive(naive_datetime, time_zone, time_zone_database) do
       {:ok, datetime} ->
         datetime
 
+      {:ambiguous, dt1, dt2} ->
+        raise ArgumentError,
+              "cannot convert #{inspect(naive_datetime)} to datetime because such " <>
+                "instant is ambiguous in time zone #{time_zone} as there is an overlap " <>
+                "between #{inspect(dt1)} and #{inspect(dt2)}"
+
+      {:gap, dt1, dt2} ->
+        raise ArgumentError,
+              "cannot convert #{inspect(naive_datetime)} to datetime because such " <>
+                "instant does not exist in time zone #{time_zone} as there is a gap " <>
+                "between #{inspect(dt1)} and #{inspect(dt2)}"
+
       {:error, reason} ->
         raise ArgumentError,
-              "cannot parse #{inspect(naive_datetime)} to datetime, reason: #{inspect(reason)}"
+              "cannot convert #{inspect(naive_datetime)} to datetime, reason: #{inspect(reason)}"
+    end
+  end
+
+  @doc """
+  Changes the time zone of a `DateTime`.
+
+  Returns a `DateTime` for the same point in time, but instead at
+  the time zone provided. It assumes that `DateTime` is valid and
+  exists in the given timezone and calendar.
+
+  By default, it uses the default time_zone returned by
+  `Calendar.get_time_zone_database/1`, which defaults to
+  `Calendar.UTCOnlyTimeZoneDatabase` which only handles "Etc/UTC" datetimes.
+  Another time zone database can be passed as argument or set globally.
+  See the "Time zone database" section in the module docs.
+
+  ## Examples
+
+      iex> cph_datetime = DateTime.from_naive!(~N[2018-07-16 12:00:00], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      iex> {:ok, pacific_datetime} = DateTime.shift_zone(cph_datetime, "America/Los_Angeles", FakeTimeZoneDatabase)
+      iex> pacific_datetime
+      #DateTime<2018-07-16 03:00:00-07:00 PDT America/Los_Angeles>
+
+  """
+  @doc since: "1.8.0"
+  @spec shift_zone(t, Calendar.time_zone(), Calendar.time_zone_database()) ::
+          {:ok, t} | {:error, :time_zone_not_found | :utc_only_time_zone_database}
+  def shift_zone(datetime, time_zone, time_zone_database)
```

Comment:
The documentation says that the `time_zone_database` defaults to one returned by `Calendar.get_time_zone_database/1`, so I think the default value is missed here.